### PR TITLE
[fix] Ensure sunburst column ordering adheres to hierarchy

### PR DESCRIPTION
**Pull request by john-bodley**
### CATEGORY

Choose one

- [x] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

This PR fixes an issue where the values for the sunburst chart (which takes in a list of lists order by the hierarchical columns) were being named incorrectly. The reason being is the resulting `pandas.DataFrame` column ordering is not the same as the ordering defined in the hierarchy. 

The fix is to simply re-order the `pandas.DataFrame` columns to adhere to the column ordering specified in the hierarchy. 

### TEST PLAN

Tested locally for a chart where there were more than one hierarchies defined and confirmed that the ordering of the layers in the chart were correct.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS

to: @etr2460 @mistercrunch @williaster

### CODE REVIEW

File: superset/viz.py [hunk 1]
```diff
@@ -1601,13 +1601,18 @@ class SunburstViz(BaseViz):
 
     def get_data(self, df: pd.DataFrame) -> VizData:
         fd = self.form_data
-        cols = fd.get("groupby")
+        cols = fd.get("groupby") or []
```
Comment (serenajiang) [inline superset/viz.py]:
```suggestion
        cols = fd.get("groupby", [])
```

File: superset/viz.py [hunk 2]
```diff
@@ -1601,13 +1601,18 @@ class SunburstViz(BaseViz):
...
-            df.columns = cols + ["m1"]  # type: ignore
+            df.rename(columns={df.columns[-1]: "m1"}, inplace=True)
             df["m2"] = df["m1"]
-        return json.loads(df.to_json(orient="values"))
+            cols.extend(["m1", "m2"])
+
+        # Re-order the columns as the query result set column ordering may differ from
+        # that listed in the hierarchy.
+        df = df[cols]
+        return df.to_numpy().tolist()
```
Comment (serenajiang) [inline superset/viz.py]: `.values.tolist()`?

Comment (john-bodley) [reply superset/viz.py]: @serenajiang per the [documentation](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.values.html#pandas.DataFrame.values) they mention, 

> Warning We recommend using DataFrame.to_numpy() instead.

**Review by serenajiang — APPROVED** [review]

**Review by williaster — COMMENTED** [review]
thanks @john-bodley !

**Pull request merged**